src/perf/lib/PerfServer.cpp

```diff
@@ -0,0 +1,280 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Server Implementation.
+
+--*/
+
+#include "PerfServer.h"
+
+#ifdef QUIC_CLOG
+#include "PerfServer.cpp.clog.h"
+#endif
+
+static
+void
+PrintHelp(
+    ) {
+    WriteOutput(
+        "\n"
+        "Perf Server options:\n"
+        "\n"
+        "  -port:<####>                The UDP port of the server. (def:%u)\n"
+        "  -selfsign:<0/1>             Uses a self-signed server certificate.\n"
+        "  -thumbprint:<cert_hash>     The hash or thumbprint of the certificate to use.\n"
+        "  -cert_store:<store name>    The certificate store to search for the thumbprint in.\n"
+        "  -machine_cert:<0/1>         Use the machine, or current user's, certificate store. (def:0)\n"
+        "\n",
+        PERF_DEFAULT_PORT
+        );
+}
+
+PerfServer::PerfServer(
+    _In_ PerfSelfSignedConfiguration* SelfSignedConfig
+    ) : SelfSignedConfig{SelfSignedConfig} {
+    if (Session.IsValid()) {
+        Session.SetAutoCleanup();
+        Session.SetPeerBidiStreamCount(PERF_DEFAULT_STREAM_COUNT);
+        Session.SetPeerUnidiStreamCount(PERF_DEFAULT_STREAM_COUNT);
+        Session.SetDisconnectTimeout(PERF_DEFAULT_DISCONNECT_TIMEOUT);
+        Session.SetIdleTimeout(PERF_DEFAULT_IDLE_TIMEOUT);
+    }
+    if (DataBufferBuffered) {
+        QUIC_FREE(DataBufferBuffered);
```

Comment:
Why are you doing this in the constructor? They're already nullptr. Did you mean this to be in a destructor?

Reply:
Oops. I meant to put it in the destructor!